tests/kernel/ee/interrupt/intc_order.cpp

```diff
@@ -0,0 +1,160 @@
+#include <common-ee.h>
+#include <kernel.h>
+#include <string.h>
+
+#define MAX_INTC_RESULTS 9
+#define MAX_INTC_HANDLERS 10
+
+static u32 g_intcSema = 0;
+static u32 g_intcHandlers[MAX_INTC_HANDLERS];
+static u32 g_intcHandlerCount = 0;
+static u32 g_orderPos = 0;
+static u32 g_order[MAX_INTC_RESULTS];
+
+typedef s32 (*IntcHandlerFunction)(s32, void *, void *);
+
+void appendResult(u32 result) {
+	if(g_orderPos != MAX_INTC_RESULTS) {
+		iSignalSema(g_intcSema);
+		g_order[g_orderPos] = result;
+		g_orderPos++;
+	}
+}
+
+s32 continueIntcHandler(s32 cause, void *arg, void *addr) {
+	appendResult((u32)arg);
+	return 0;
+}
+
+s32 cancelIntcHandler(s32 cause, void *arg, void *addr) {
+	appendResult((u32)arg);
+	return -1;
+}
+
+void initializeTest() {
+	g_orderPos = 0;
+	g_intcHandlerCount = 0;
+	{
+		ee_sema_t intcSemaInfo;
+		memset(&intcSemaInfo, 0, sizeof(intcSemaInfo));
+		intcSemaInfo.init_count = 0;
+		intcSemaInfo.max_count  = MAX_INTC_RESULTS;
+		intcSemaInfo.option     = 0;
+		g_intcSema = CreateSema(&intcSemaInfo);
+	}
+}
+
+void addTestIntcHandler(IntcHandlerFunction handler, s32 next, void *arg) {
+	int handlerId = AddIntcHandler2(INTC_VBLANK_S, handler, next, arg);
```

Comment:
Okay... I think it makes some sense to test the other set of functions elsewhere (although I personally would template it or something and test both here), but I think they should agree within this test, imho.

-[Unknown]